### PR TITLE
Gavsum/ch37466/add invert flag and redirect suffix

### PR DESCRIPTION
**Pull request by Gavsum**
* These changes were added to support functionality needed in ch37466
* Changes
    * new param: InvertFlag, when true it flips the value of the actual flag internally. Use case: redirect from route if actual flag value is true.
    * accept to param as function to allow evaluation of to value at runtime for dynamic route values
    * ldRedirect does redirect handling on mount and activate to handle keep-alive context

### CODE REVIEW

**Comment by Gavsum** [comment]
@benjamin-t-frost adding the 'semi-untestable' activated/mount keep alive handling is whats reducing the codecov. I'll leave it up to you to decide on coverage vs fuctionality.

**Comment by benjamin-t-frost** [comment]
> @benjamin-t-frost adding the 'semi-untestable' activated/mount keep alive handling is whats reducing the codecov. I'll leave it up to you to decide on coverage vs fuctionality.

I'm surprised codecov is still working 🤔

**Review by jeverick — APPROVED** [review]

File: src/mixins/ldRedirect.js [hunk 1]
```diff
@@ -53,14 +68,25 @@ export default (requiredFeatureFlag, to) => {
...
+      if (this.hasBeenDeactivated) {
+        this.redirectHandler();
       }
     },
+    mounted() {
+      this.redirectHandler();
+    },
+    deactivated() {
+      this.hasBeenDeactivated = true;
+    },
```
Comment (benjamin-t-frost) [inline src/mixins/ldRedirect.js]: A note that this is intended to support components within <keep-alive> would be nice for future context.

File: src/mixins/ldRedirect.js [hunk 2]
```diff
@@ -1,17 +1,32 @@
-export default (requiredFeatureFlag, to) => {
+export default (requiredFeatureFlag, to, invertFlag) => {
   return {
     data() {
       return {
         ldRedirectReadyWatcher: null,
         ldRedirectFlagWatcher: null,
+        hasBeenDeactivated: false,
```
Comment (benjamin-t-frost) [inline src/mixins/ldRedirect.js]: Would prefer that every property which risks colliding with a property in the host component be prefixed with `ldRedirect`

(recurs)

**Review by benjamin-t-frost — APPROVED** [review]

**Review by kcaseygover — APPROVED** [review]

**Pull request merged**